Actions/PrimaryGeneratorAction.h

```diff
@@ -70,7 +70,7 @@ class PrimaryGeneratorAction : public G4VUserPrimaryGeneratorAction
 
 private:
   G4String fGenerateSourceType;
-  G4String fAllowedSourceTypes[4] = {"run", "beam", "isotope", "nema"};
+  G4String fAllowedSourceTypes[5] = {"run", "beam", "isotope", "nema", "cosmics"};
```

Comment:
could it be just `std::vector<G4String> fAllowedSourceTypes = = {"run", "beam", "isotope", "nema", "cosmics"};`
Then you don't need to explictly say how many elements you have

Reply:
applied